sustainability/models/carbon_factor.py

```diff
@@ -493,12 +494,32 @@ def _get_carbon_value(
                 monetary_currency_id,
             ) = factor_value.get_infos()
 
+            weight_uom_category = self.env.ref("uom.product_uom_categ_kgm")
+
             if compute_method == "monetary" and amount is not None and from_currency_id:
                 # We convert the amount to the currency used in the factor value
                 partial_value_result = carbon_value * from_currency_id._convert(
                     amount, monetary_currency_id, self.env.company, date
                 )
-
+            elif (
+                compute_method == "physical"
+                and quantity is not None
+                and self.carbon_uom_id.category_id == weight_uom_category
+            ):
+                if not product_id.weight or product_id.weight <= 0:
+                    raise ValidationError(
+                        _(
+                            "The weight may not be defined or is zero for the associated product (%s). "
+                            "Please ensure the weight is properly set to compute the carbon value.",
+                            product_id.display_name,
+                        )
+                    )
+                default_weight_uom = self.env.ref("uom.product_uom_kgm")
```

Comment:
it's not necessary product_uom_kgm , some countries use product_uom_lb
see _get_weight_uom_id_from_ir_config_parameter